Formula/lib/libxml2.rb

```diff
@@ -37,7 +38,7 @@ class Libxml2 < Formula
   depends_on "python@3.11" => [:build, :test]
   depends_on "python@3.12" => [:build, :test]
   depends_on "pkg-config" => :test
-  depends_on "icu4c"
+  depends_on "icu4c@75"
```

Comment:
This may cause issues on Linux due to heavy use of `uses_from_macos "libxml2"` (e.g. `swift` will have multiple `icu4c` deps). It depends on how formulae behave when a dependency is running a different `icu4c` version.

Reply:
> how formulae behave when a dependency is running a different `icu4c` version.

Very poorly, most likely

Reply:
> > how formulae behave when a dependency is running a different `icu4c` version.
> 
> Very poorly, most likely

In the case that a formula isn't actually using `icu4c` feature of a dependency, then it may work. However, this is hard to guarantee and there are also potential problems when `dlopen()` is used.

---

Another tricky thing is users will have trouble running `pkg-config` when `icu4c` is keg-only. This means manually modifying all pc files (which I was trying out locally but becoming a bit of a pain).

And dependents will grab those flags from pkg-config or dedicated config scripts like
```console
❯ /opt/homebrew/opt/libxml2/bin/xml2-config --cflags --libs
-I/opt/homebrew/Cellar/libxml2/2.12.6_1/include/libxml2 -L/opt/homebrew/Cellar/libxml2/2.12.6_1/lib -lxml2 -lz -L/opt/homebrew/Cellar/icu4c@75/75.1/lib -licui18n -licuuc -licudata -liconv
```

So may lead to build-time issues from mixing headers and/or libs.

---

I think there are enough problems that we won't be able to use this approach. May just do the versioned `icu4c@74` for any formula that doesn't build with new ICU or with C++17.